build.cake

```diff
@@ -18,8 +18,6 @@ var ErrorDetail = new List<string>();
 var version = "3.6.0";
 var modifier = "";
 
-var isCompactFrameworkInstalled = FileExists(Environment.GetEnvironmentVariable("windir") + "\\Microsoft.NET\\Framework\\v3.5\\Microsoft.CompactFramework.CSharp.targets");
-
 //Find program files on 32-bit or 64-bit Windows
 var programFiles = Environment.GetEnvironmentVariable("ProgramFiles(x86)") ?? Environment.GetEnvironmentVariable("ProgramFiles");
 var isSilverlightSDKInstalled = FileExists(programFiles  + "\\MSBuild\\Microsoft\\Silverlight\\v5.0\\Microsoft.Silverlight.CSharp.targets");
```

Comment:
This could either be a task or an inline check at the start of the script. I lean toward  the inline check (which could be a method call) if __all__ tasks would have to depend on the Check task. A task is better if only some tests require it. Using SetUp is another option, and now that I think of it probably cleanest.